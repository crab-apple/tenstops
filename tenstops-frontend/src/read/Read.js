import React from 'react'
import PropTypes from 'prop-types'
import ReadShape from './ReadShape'
import { Loader, Header } from 'semantic-ui-react'

const Read = ({read, content}) => {
  if (read && content) {
    return (
      <div>
        <Header>
          {read.title}
          <Header.Subheader>
            by {read.author}
          </Header.Subheader>
        </Header>
        <Paragraphs text={content} />
      </div>
    )
  } else {
    return <Loader active />
  }
}

const Paragraphs = ({text}) => {
  let parCounter = 1
  const paragraphs = text.split('\n')
    .map((p) => p.trim())
    .filter((p) => p.length)
  return (
    <div>
      {paragraphs.map((p) => <p key={parCounter++}>{p}</p>)}
    </div>
  )
}

Read.propTypes = {
  read: ReadShape,
  content: PropTypes.string
}

export default Read
