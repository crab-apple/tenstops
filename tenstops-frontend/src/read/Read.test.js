/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Read from './Read'

test('snapshot', () => {
  const read = {
    id: 1,
    title: 'Novel 1',
    author: 'John Doe 1'
  }

  const content = 'foo bar bar \n\n bar bar foo'

  const component = renderer.create(
    <Read read={read} content={content} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
