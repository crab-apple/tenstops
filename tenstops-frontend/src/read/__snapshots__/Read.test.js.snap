// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    className="ui header"
  >
    Novel 1
    <div
      className="sub header"
    >
      by 
      John Doe 1
    </div>
  </div>
  <div>
    <p>
      foo bar bar
    </p>
    <p>
      bar bar foo
    </p>
  </div>
</div>
`;
